doc/priorityClass.md

```diff
@@ -0,0 +1,46 @@
+# PriorityClass proposed flow design
+
+This page describes the design and implementation of the PriorityClass
+support in the compliance operator.
+
+## Goals
+We want to accomplish:
+   * User can set PriorityClass in ScanSetting
+   * Pods will be launched with the PriorityClass set in ScanSetting
+
+## Add PriorityClass option to ScanSetting CRD
+
+Administrators can choose which PriorityClass in the ScanSetting.
+A sample looks as follows:
+
+```yaml
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: my-companys-constraints
+autoApplyRemediations: false
+autoUpdateRemediations: false
+schedule: "0 1 * * *"
+priorityClass: my-priority-class
+rawResultStorage:
+  size: "2Gi"
+  rotation: 10
+# For each role, a separate scan will be created pointing
+# to a node-role specified in roles
+roles:
+  - worker
+  - master
+```
+
+We will be adding priorityClass to ScanSetting CRD as a optional field,
+and if none is specified, we will treat as it didn't have a PriorityClass.
+Otherwise, we will use the PriorityClass specified in the ScanSetting.
+
+## Launch related scanning pod with set PriorityClass in ScanSetting
+
+Once an admin set a PriorityClass in ScanSetting, when this ScanSetting
```

Comment:
Yes, I think anyone who has the ability to create a ScanSetting should be able to set the PriorityClass

Reply:
This was bothering me for a bit but then I realized it's not that big of a deal because:
1) creating the scans, even without remediations already gives you a lot of sensitive information
2) we ship a default-auto-apply setting that can even change the cluster.
So there's not much more going on if the admin can also set the priority class.